tests/invoice.spec.js

```diff
@@ -72,6 +72,60 @@ describe('Invoice', function () {
       it('should have `tetelek` node', function () {
         expect(obj).to.have.property('tetelek')
       })
-    })
-  })
-})
+    
+      // START- New test suite for adjustmentInvoiceNumber property
+      describe('adjustmentInvoiceNumber validation', function () {
+        it('should not include adjustmentInvoiceNumber when it is null', function (done) {
+          invoice._options.adjustmentInvoiceNumber = null;
+          parser.parseString('<wrapper>' + invoice._generateXML() + '</wrapper>', function (err, result) {
+            expect(result.wrapper).to.not.have.deep.property('fejlec.helyesbitettSzamlaszam');
+            done();
+          });
+        });
+
+        it('should not include adjustmentInvoiceNumber when it is undefined', function (done) {
+          delete invoice._options.adjustmentInvoiceNumber;
+          parser.parseString('<wrapper>' + invoice._generateXML() + '</wrapper>', function (err, result) {
+            expect(result.wrapper).to.not.have.deep.property('fejlec.helyesbitettSzamlaszam');
+            done();
+          });
+        });
+
+        it('should throw an error when adjustmentInvoiceNumber is an empty string', function () {
+          expect(() => {
+            invoice._options.adjustmentInvoiceNumber = '';
+            invoice._generateXML();
+          }).to.throw();
+        });
+
+        it('should throw an error when adjustmentInvoiceNumber is not a string', function () {
+          const invalidTypes = [new Date(), 123, true];
```

Comment:
it is nice simplification, but if it fails not easy to debug what was wrong.
There are 2 options.

Dynamically create test `it`
```javascript
const invalidTypes = [new Date(), 123, true];
invalidTypes.forEach(type => {
  it(`should throw an error when adjustmentInvoiceNumber is ${typeof type}`, function () {
    expect(() => {
    invoice._options.adjustmentInvoiceNumber = type;
    invoice._generateXML();
    }).to.throw(/"adjustmentInvoiceNumber" should be a string/);
  });
});
```

Create 3 independent test

My problem with the 1st approach is
- `typeof new Date` is `object`. That logically good, but 
- you can't test/debug only 1 test if fail.
- I don't like overcomplicate test. It should have minimal logic. I also don't like code duplication but code duplication in test are not same as in code. It is my personal preference, but many times when the code changed I did not have to rewrite the "testing world"

So I vote on the 2nd solution, but I accept the dynamic version too.

Reply:
Thank you very much for your time and the valuable suggestions for testing. It was great consulting and discussing these matters with you. I found it very beneficial and I agree with your recommendations. I have endeavored to implement them as expected.